test/utility.cpp

```diff
@@ -17,6 +17,81 @@
 #define CATCH_CONFIG_MAIN
 #include "catch.hpp"
 
+#define CHECK_CXX14_CONSTEXPR(Tag, ...)                                        \
+  struct check_cxx14_constexpr_helper_##Tag {                                  \
+    static SLB_CXX14_CONSTEXPR bool check() { __VA_ARGS__ return true; };      \
```

Comment:
I feel like we should leave the `return true;` to be part of the `__VA_ARGS__`, and the test case can
explicitly say `return true;` if that's actually what's desired. At least in the one case we use below,
we have `return old_obj == 42 && obj == 43;` which overrides the `return true;`. I would think that
most test cases we'd have a value we explicitly want to return/check?